src/extension/features/general/colour-blind-mode/index.js

```diff
@@ -1,7 +1,356 @@
+  var g;
+  var b;
+
+  x = 0.95047 * (x * x * x > 0.008856 ? x * x * x : (x - 16 / 116) / 7.787);
+  y = 1.0 * (y * y * y > 0.008856 ? y * y * y : (y - 16 / 116) / 7.787);
+  z = 1.08883 * (z * z * z > 0.008856 ? z * z * z : (z - 16 / 116) / 7.787);
+
+  r = x * 3.2406 + y * -1.5372 + z * -0.4986;
+  g = x * -0.9689 + y * 1.8758 + z * 0.0415;
+  b = x * 0.0557 + y * -0.204 + z * 1.057;
+
+  r = r > 0.0031308 ? 1.055 * Math.pow(r, 1 / 2.4) - 0.055 : 12.92 * r;
+  g = g > 0.0031308 ? 1.055 * Math.pow(g, 1 / 2.4) - 0.055 : 12.92 * g;
+  b = b > 0.0031308 ? 1.055 * Math.pow(b, 1 / 2.4) - 0.055 : 12.92 * b;
+
+  return [
+    Math.max(0, Math.min(1, r)) * 255,
+    Math.max(0, Math.min(1, g)) * 255,
+    Math.max(0, Math.min(1, b)) * 255,
+  ];
+}
+
+function rgb2lab(rgb) {
+  var r = rgb[0] / 255;
+  var g = rgb[1] / 255;
+  var b = rgb[2] / 255;
+  var x;
+  var y;
+  var z;
+
+  r = r > 0.04045 ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92;
+  g = g > 0.04045 ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92;
+  b = b > 0.04045 ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92;
+
+  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047;
+  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.0;
+  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883;
+
+  x = x > 0.008856 ? Math.pow(x, 1 / 3) : 7.787 * x + 16 / 116;
+  y = y > 0.008856 ? Math.pow(y, 1 / 3) : 7.787 * y + 16 / 116;
+  z = z > 0.008856 ? Math.pow(z, 1 / 3) : 7.787 * z + 16 / 116;
+
+  return [116 * y - 16, 500 * (x - y), 200 * (y - z)];
+}
+
+/* END OF LICENSE */
+
+function hex2rgb(hex) {
+  if (hex.length !== 7) {
+    return [0, 0, 0];
+  }
+
+  var r = parseInt(hex.slice(1, 3), 16);
+  var g = parseInt(hex.slice(3, 5), 16);
+  var b = parseInt(hex.slice(5, 7), 16);
+
+  return [Number.isNaN(r) ? 0 : r, Number.isNaN(g) ? 0 : g, Number.isNaN(b) ? 0 : b];
+}
+
+function rgb2hex(rgb) {
+  var r = Math.round(rgb[0]).toString(16);
+  var g = Math.round(rgb[1]).toString(16);
+  var b = Math.round(rgb[2]).toString(16);
+
+  return (
+    '#' +
+    (r.length === 2 ? r : '0' + r) +
+    (g.length === 2 ? g : '0' + g) +
+    (b.length === 2 ? b : '0' + b)
+  );
+}
+
+function lab2lch(lab) {
+  var c = Math.sqrt(lab[1] * lab[1] + lab[2] * lab[2]);
+  var h = Math.atan2(lab[2], lab[1]);
+  if (h > 0) {
+    h = (h / Math.PI) * 180;
+  } else {
+    h = 360 - (Math.abs(h) / Math.PI) * 180;
+  }
+  return [lab[0], c, h];
+}
+
+function lch2lab(lch) {
+  var a = Math.cos((lch[2] / 180) * Math.PI) * lch[1];
+  var b = Math.sin((lch[2] / 180) * Math.PI) * lch[1];
+  return [lch[0], a, b];
+}
+
+function hexToLch(hex) {
+  return lab2lch(rgb2lab(hex2rgb(hex)));
+}
+
+function lchToHex(l, c, h) {
+  return rgb2hex(lab2rgb(lch2lab([l, c, h])));
+}
 
 export class ColourBlindMode extends Feature {
   injectCSS() {
     return require('./index.css');
   }
+
+  shouldInvoke() {
+    var optionMenu = $(`.${TK_COLOUR_BLIND_OPTION_MENU}`);
+    return optionMenu.length === 0;
+  }
+
+  calculateAccents(hex) {
```

Comment:
Instead of just appending the `lc` name, can we humanize it just a bit to something like:

```css
--tk-colour-blind__negative--dark-3
--tk-colour-blind__negative--dark-2
--tk-colour-blind__negative--dark-1
--tk-colour-blind__negative--base
--tk-colour-blind__negative--light-1
--tk-colour-blind__negative--light-2
--tk-colour-blind__negative--light-3
```

Just so it's a bit easier to know what's used in the `.css` file?

Reply:
I'm happy to - I wasn't sure how to order them though. I could order them lightest to darkest or vice versa but if we needed to introduce another accent colour later on we'd have to shift them all down a rung in the CSS to keep them in order. 
Alternatively, not having them ordered seemed less useful than having the l/c values to compare.

Reply:
in my experience, it's rare colors get added to palettes so it's not a big deal. that said, just add a 0. so `base`, `10`, `20`, `30`, etc.

Reply:
I've thought about it more and I'm kinda over it haha. Let's leave these as is but can you add comments to the accents in code that:

1. Link to a place to easily come up with new accents.
2. Distinguish the accents by name or numerical value.

That way it's still documented that somewhere which value is lighter/darker but also easier for us to add new ones as needed in the future.